File: back/src/main/java/com/souzs/back/architecture_auth_examples_back/domain/stock/entity/Stock.java
package com.souzs.back.architecture_auth_examples_back.domain.stock.entity;

import com.souzs.back.architecture_auth_examples_back.domain.catalog.entity.Product;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;

@Entity
@Table(name = "stocks")
@Getter
@Setter
@NoArgsConstructor
public class Stock {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "product_id", nullable = false, unique = true)
    private Product product;

    @Column(nullable = false)
    private Integer quantity = 0;

    @Column(nullable = false)
    private Integer minQuantity = 0;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private StockStatus status = StockStatus.PENDING_COUNT;

    @Column(nullable = false)
    private Instant updatedAt;

    @PrePersist
    @PreUpdate
    protected void onUpdate() {
        updatedAt = Instant.now();
    }
}

File: front/src/models/stock.ts
export type StockStatus = 'REGULAR' | 'DIVERGENT' | 'PENDING_COUNT' | 'LOW_STOCK'

export interface Stock {
  id: number
  productId: number
  productName: string
  quantity: number
  minQuantity: number
  status: StockStatus
  updatedAt: string
}

export interface StockCount {
  id: number
  stockId: number
  productName: string
  countedByUserId: number
  countedByName: string
  countedQuantity: number
  countedAt: string
}
